dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs

```diff
@@ -73,7 +83,7 @@ getLetIdentifier src@(Src left _ text) =
   where parseLetIdentifier = do
           setSourcePos left
           _let
-          whitespace
+          nonemptyWhitespace
```

Comment:
Why do we actually have all this duplicated parsing logic? Is it impossible to share this with `dhall`'s parser? Could you maybe add some documentation for this @EggBaconAndSpam?

Reply:
I believe the problem was that the parser is parameterised by the parser for import expressions, which means we can't easily export its components. I'll have to think about it. I admit that the current situation is a bit of a hack.

Reply:
@EggBaconAndSpam: You can import export additional parsers if you need to by reusing the same trick we use for exporting the parser for `importExpression`:

https://github.com/dhall-lang/dhall-haskell/blob/53320c8e1944c9c8d72c380fe1232185aa03b276/dhall/src/Dhall/Parser/Expression.hs#L105-L125

Reply:
I think I'll look into that separately. It seems there are a few more bugs that recently crept in that I want to look at.